### PR TITLE
Promote dev dependencies action

### PR DESCRIPTION
**Pull request by homapf**

### CODE REVIEW

File: .github/actions/promote-dev-dependencies/promote-dev-dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Function to process a single package.json file
+process_package_json() {
+    local file="$1"
+    local homa_dependencies_key="$2"
```
Comment (manelizzard) [inline .github/actions/promote-dev-dependencies/promote-dev-dependencies.sh:6]: Can we generalize this variable name?

File: .github/actions/promote-dev-dependencies/promote-dev-dependencies.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        else
+            echo "No $homa_dependencies_key found in $file"
+        fi
+    fi
+}
+
+# Main function to process all package.json files in a folder and its subfolders
+process_all_package_json() {
+    local folder="$1"
+    local homa_dependencies_key="$2"
```
Comment (manelizzard) [inline .github/actions/promote-dev-dependencies/promote-dev-dependencies.sh:27]: Same here

**Review by manelizzard — CHANGES_REQUESTED** [review]
Asking to generalize `homa_dependencies....` variables

File: .github/actions/promote-dev-dependencies/promote-dev-dependencies.sh (hunk 1 above)
Comment (homapf) [reply .github/actions/promote-dev-dependencies/promote-dev-dependencies.sh]: Done

File: .github/actions/promote-dev-dependencies/promote-dev-dependencies.sh (hunk 2 above)
Comment (homapf) [reply .github/actions/promote-dev-dependencies/promote-dev-dependencies.sh]: Done

**Review by manelizzard — APPROVED** [review]

**Pull request merged**